src/test/test-path-util.c

```diff
@@ -221,18 +226,32 @@ static void test_find_executable_full(void) {
 
         assert_se(unsetenv("PATH") == 0);
 
-        assert_se(find_executable_full("sh", NULL, true, &p, NULL) == 0);
+        assert_se(find_executable_full("sh", NULL, NULL, true, &p, NULL) == 0);
         puts(p);
         assert_se(streq(basename(p), "sh"));
         free(p);
 
-        assert_se(find_executable_full("sh", NULL, false, &p, NULL) == 0);
+        assert_se(find_executable_full("sh", NULL, NULL, false, &p, NULL) == 0);
         puts(p);
         assert_se(streq(basename(p), "sh"));
         free(p);
 
         if (oldpath)
                 assert_se(setenv("PATH", oldpath, true) >= 0);
+
+        _cleanup_(unlink_tempfilep) char fn[] = "/tmp/test-XXXXXX";
+        fd = mkostemp_safe(fn);
+        fchmod(fd, 0755);
+
+        test_file_name = strchr(strchr(fn, delim), delim);
```

Comment:
the outer strchr() will return the same character the first strchr() already found